lib/matplotlib/_types.py

```diff
@@ -0,0 +1,165 @@
+"""
+Style desription information that is shared across unrelated classses.
+"""
+
+from enum import Enum
+from matplotlib import cbook
+
+
+def _deprecate_case_insensitive_join_cap(s):
+    s_low = s.lower()
+    if s != s_low:
+        if s_low in ['miter', 'round', 'bevel']:
+            cbook.warn_deprecated(
+                "3.3", message="Case-insensitive capstyles are deprecated "
+                "since %(since)s and support for them will be removed "
+                "%(removal)s; please pass them in lowercase.")
+        elif s_low in ['butt', 'round', 'projecting']:
+            cbook.warn_deprecated(
+                "3.3", message="Case-insensitive joinstyles are deprecated "
+                "since %(since)s and support for them will be removed "
+                "%(removal)s; please pass them in lowercase.")
+        # Else, error out at the check_in_list stage.
+    return s_low
```

Comment:
Deprecation machinery copied as-is out of `rcsetup.py`.

Reply:
This seems arbitrary, and should definitely be a class method if you really want to keep it in.  Why would we care if join styles are capitalized or not?  Do we really think "round" and "Round" will ever mean something different?

Reply:
This is existing behavior. I was not around for the discussion about whether or not to allow caps. I agree that for a user-facing thing, allowing case-insensitivity is probably fine, but this should probably not be hashed out in this PR.